nntrainer/tensor/uint_tensor.cpp

```diff
@@ -1,46 +1,54 @@
 // SPDX-License-Identifier: Apache-2.0
 /**
- * @file	short_tensor.cpp
+ * @file	uint_tensor.cpp
  * @date	02 April 2024
- * @brief	This is ShortTensor class for 16-bit unsigned integer calculation
+ * @brief	This is UIntTensor class for unsigned integer calculation
  * @see		https://github.com/nnstreamer/nntrainer
  * @author	Donghyeon Jeong <dhyeon.jeong@samsung.com>
+ * @author	Eunju Yang <ej.yang@samsung.com>
  * @bug		No known bugs except for NYI items
  */
 
 #include <iomanip>
 #include <iostream>
 
 #include <blas_interface.h>
-#include <short_tensor.h>
 #include <tensor.h>
+#include <uint_tensor.h>
 
 namespace nntrainer {
 
-ShortTensor::ShortTensor(std::string name_, Tformat fm) :
-  TensorBase(name_, fm, Tdatatype::UINT16) {}
+template class UIntTensor<uint8_t, Tdatatype::UINT8>;
+template class UIntTensor<uint16_t, Tdatatype::UINT16>;
+template class UIntTensor<uint32_t, Tdatatype::UINT32>;
 
-ShortTensor::ShortTensor(const TensorDim &d, bool alloc_now, Initializer init,
-                         std::string name) :
+template <typename T, Tdatatype D>
+UIntTensor<T, D>::UIntTensor(std::string name_, Tformat fm) :
+  TensorBase(name_, fm, D) {}
```

Comment:
It seems like `Tdatatype D` is used only in constructors. I think it could be removed.

```suggestion
template <typename T>
UIntTensor<T>::UIntTensor(std::string name_, Tformat fm) :
  TensorBase(name_, fm, checkTensorDataType()) {}
```

and define a private function `checkTensorDataType()`.
```
// uint_tensor.h
Tdatatype checkTensorDataType() const {
  if (typeid(T) == typeid(uint8_t))
    return Tdatatype::UINT8;
  else if (typeid(T) == typeid(uint16_t))
    return Tdatatype::UINT16;
  else if (typeid(T) == typeid(uint32_t))
    return Tdatatype::UINT32;
  else
    throw std::runtime_error("unsupported type");
}
```

Reply:
Thank you for your suggestion. It sounds great! It seems much better to update the code like you recommended. I will update the code.